src/java/org/apache/cassandra/service/DefaultFSErrorHandler.java

```diff
@@ -86,6 +86,9 @@ public void handleFSError(FSError e)
                         Keyspace.removeUnreadableSSTables(directory);
                 }
                 break;
+            case die:
```

Comment:
This feels like the wrong place to have a fix like this. This also applies to the bloom filter tracking changes and anything else that is a valuable fix on its own.
Do we have separate tickets for them? Let's make sure we have and prioritize them so that they are committed to OSS before we propose this. (To clarify, I do not want to remove any of these from this patch.)